sdk/eventhub/event-hubs/test/client.spec.ts

```diff
@@ -152,149 +286,244 @@ describe("ServiceCommunicationError for non existent namespace", function(): voi
   });
 });
 
-describe("MessagingEntityNotFoundError for non existent eventhub", function(): void {
-  let client: EventHubClient;
+describe("EventHubConsumerClient with non existent event hub", function(): void {
+  let client: EventHubConsumerClient;
+  const expectedErrCode = "MessagingEntityNotFoundError";
 
   beforeEach(() => {
     should.exist(
       env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
       "define EVENTHUB_CONNECTION_STRING in your environment before running integration tests."
     );
-    client = new EventHubClient(env[EnvVarKeys.EVENTHUB_CONNECTION_STRING], "bad" + Math.random());
+
+    client = new EventHubConsumerClient("dummy", env[EnvVarKeys.EVENTHUB_CONNECTION_STRING], "bad");
   });
 
   afterEach(() => {
     return client.close();
   });
 
-  it("should throw MessagingEntityNotFoundError while getting hub runtime info", async function(): Promise<
+  it("should throw MessagingEntityNotFoundError for getEventHubProperties", async function(): Promise<
     void
   > {
     try {
-      await client.getProperties();
+      await client.getEventHubProperties();
       throw new Error("Test failure");
     } catch (err) {
       debug(err);
-      should.equal(err.code, "MessagingEntityNotFoundError");
+      should.equal(err.code, expectedErrCode);
     }
   });
 
-  it("should throw MessagingEntityNotFoundError while getting partition runtime info", async function(): Promise<
+  it("should throw MessagingEntityNotFoundError for getPartitionProperties", async function(): Promise<
     void
   > {
     try {
       await client.getPartitionProperties("0");
       throw new Error("Test failure");
     } catch (err) {
       debug(err);
-      should.equal(err.code, "MessagingEntityNotFoundError");
+      should.equal(err.code, expectedErrCode);
     }
   });
 
-  it("should throw MessagingEntityNotFoundError while creating a sender", async function(): Promise<
+  it("should throw MessagingEntityNotFoundError for getPartitionIds", async function(): Promise<
     void
   > {
     try {
-      const sender = client.createProducer({ partitionId: "0" });
-      await sender.send([{ body: "Hello World" }]);
+      await client.getPartitionIds();
       throw new Error("Test failure");
     } catch (err) {
       debug(err);
-      should.equal(err.code, "MessagingEntityNotFoundError");
+      should.equal(err.code, expectedErrCode);
     }
   });
 
-  it("should throw MessagingEntityNotFoundError while creating a receiver", async function(): Promise<
+  it("should throw MessagingEntityNotFoundError while subscribe()", async function(): Promise<
+    void
+  > {
+    let caughtErr: any;
+    const subscription = client.subscribe({
+      processEvents: async () => {},
+      processError: async (err) => {
+        caughtErr = err;
+      }
+    });
+    await delay(5000);
+    await subscription.close();
```

Comment:
Minor suggestion, but we've been trying to reduce our reliance on 'magic' timeouts when possible since they end up either causing CI to take longer to run, or cause tests to be flaky.

A pattern we've been using is to wrap the call in a promise, then await that promise. So for this case, we'd change it to something like:
```typescript
    const waitForError: new Promise((resolve) => {
      const subscription = client.subscribe({
        processEvents: async () => {},
        processError: async (err) => {
          await subscription.close(); // not really needed since client.close() should also close the subscription
          resolve(err);
        }
      });
    });
    const caughtErr = await waitForError;
```

Reply:
Updated. Some observations
- subscription.close() is needed otherwise the test would wait for a minute or so to exit
- doing subscription.close() inside the error handler resulted in the test never exiting. Need to dig into that, I'll log an issue soon. For now I have moved subscription.close() to after the waitForError promise is resolved

Reply:
See https://github.com/Azure/azure-sdk-for-js/pull/9200/commits/c432e8148c5b5f7338cc7935529473c597eaca09